pkg/bpf/bpffs_migrate.go

```diff
@@ -90,35 +92,52 @@ func repinMap(bpffsPath string, name string, spec *ebpf.MapSpec) error {
 	if err != nil {
 		return fmt.Errorf("map not found at path %s: %v", name, err)
 	}
+	defer pinned.Close()
 
 	if pinned.Type() == spec.Type &&
 		pinned.KeySize() == spec.KeySize &&
 		pinned.ValueSize() == spec.ValueSize &&
 		pinned.Flags() == spec.Flags &&
 		pinned.MaxEntries() == spec.MaxEntries {
-		return nil
+		// cilium_calls_xdp is shared between XDP interfaces and should only be
```

Comment:
Could you create a GH issue to document the fact that the `cilium_calls_xdp` is shared among all XDP progs. I think it's a ticking bomb once we move more functionality to the XDP layer.

Reply:
I was already working on per-device bpffs dirs before picking up this issue, but with https://github.com/cilium/cilium/issues/20691 closed there's indeed nothing left to track. Created https://github.com/cilium/cilium/issues/29047 for this.